src/components/Integrations/BigQuery/ControlPanel/MultiInputContainer.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useIntl } from 'react-intl';
+import Input from './Input';
+
+type InputConfig = {
+  value: string;
+  id: string;
+  className?: string;
+  errorMessage?: string;
+};
+
+const uniqueId = function (name: String) {
+  return (
+    name +
+    '-xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+      let r = (Math.random() * 16) | 0,
+        v = c === 'x' ? r : (r & 0x3) | 0x8;
+      return v.toString(16);
+    })
+  );
+};
+
+const emptyInput = (): InputConfig => ({
+  id: uniqueId('input'),
+  value: '',
+  className: 'style_empty',
+  errorMessage: '',
+});
+
+const generateInputConfig = (values: string[]): InputConfig[] => {
+  return values.map((value) => {
+    return {
+      id: uniqueId('input'),
+      value,
+      className: value !== '' ? 'style_not_empty' : 'style_empty',
+    };
+  });
+};
+
+interface MultiInputProps {
+  values?: string[];
+  onChange: (newValues: string[]) => void;
+  onRemove: () => void;
+  onAdd: () => void;
+}
+
+const MultiInput = (props: MultiInputProps) => {
+  const intl = useIntl();
+  const i18 = useCallback(
+    (id: string, values?: any) => {
+      return intl.formatMessage({ id: id }, values);
+    },
+    [intl],
+  );
```

Comment:
If I am not wrong, in general, we call this function `_`, right?

Reply:
yes, an example: `const _ = (id, values) => intl.formatMessage({ id: id }, values);`

Reply:
It is necessary to use `useCallback` here?